TUS to handle lo upload

AdityaKBhadragond14 opened the pull request:
Added TUS functionality to handle LO files upload.

AdityaKBhadragond14 committed:
wip uploading lo files using tus.

AdityaKBhadragond14 committed:
TUS Service modified to upload LO files.

AdityaKBhadragond14 committed:
removed the configuration in CustomSecurityConfigNoKC

AdityaKBhadragond14 committed:
removed the configuration in CustomSecurityConfigNoKC

AdityaKBhadragond14 committed:
removed the configuration in CustomSecurityConfigNoKC

AdityaKBhadragond14 committed:
revert

joiskash commented on src/main/resources/application.yaml:
```diff
@@ -75,8 +75,9 @@ hapi:
-#      hibernate.cache.use_minimal_puts: false
+      #      hibernate.cache.use_minimal_puts: false
       image_path: image
+      hyper_spectral_files_path: hyperSpectralFiles
```
hyper-spectral-files, or hyper_spectral_files since its a folder name

joiskash reviewed (approved):
Approving for now. 
Please re-write the code to use strategy pattern to handle various file types.
Please ensure that this can be a library / Service that can be imported by any spring boot server.
The storage location should also be a stratgey eg: You can store on disk in file or in AWS S3 or in Azure blob storage etc.

AdityaKBhadragond14 committed:
Code Enhancement and Refactoring.

joiskash reviewed (approved):
Looks good, just add a context object that will manage the strategy types. Tus service will call context.setStrategy(FileType) and then context.transferFile etc,

AdityaKBhadragond14 committed:
Strategy context class.

AdityaKBhadragond14 commented:
> Looks good, just add a context object that will manage the strategy types. Tus service will call context.setStrategy(FileType) and then context.transferFile etc,

Added a context class to set the strategy and moved the other functions to it. Only cannot move the function to transfer the file because that is being used in the controller as well so it is better to keep it as part of the Tus service class.
